cmd/mysterium_client/run/command.go

```diff
@@ -52,11 +55,25 @@ func NewCommandWith(
 	tequilapi_endpoints.AddRoutesForProposals(router, mysteriumClient)
 
 	httpApiServer := tequilapi.NewServer(options.TequilapiAddress, options.TequilapiPort, router)
-
-	return &CommandRun{
+	cmd := &CommandRun{
 		connectionManager,
 		httpApiServer,
 	}
+	sigterm := make(chan os.Signal, 2)
+	signal.Notify(sigterm, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
+	go sarahConnor(sigterm, cmd)
```

Comment:
Factory should be responsible 1 responsibility - for factoring, but not running

Reply:
alright, this is now in Run()